[Build] Compile with the latest Dockfile.cuda, pass in ARG CUDA_VERSION=11.8.0 ARG CUDNN_VERSION=8.6.0.163 ARG OS=ubuntu22.04

### Describe the issue

[Build] Compile with the latest Dockfile.cuda, pass in ARG CUDA_VERSION=11.8.0 ARG CUDNN_VERSION=8.6.0.163 ARG OS=ubuntu22.04.



### Urgency

_No response_

### Target platform

CUDA x86 ubuntu2204

### Build script

The command '/bin/sh -c cd /code     && python3 -m venv /code/env     && . /code/env/bin/activate     && pip install --upgrade psutil setuptools wheel packaging     && pip install -r /code/tools/ci_build/github/linux/python/requirements.txt     && python /code/tools/ci_build/build.py --build_dir /code/build/Linux     --allow_running_as_root --skip_submodule_sync     --use_cuda --cuda_home /usr/local/cuda     --cudnn_home /code/build/cudnn     --build_shared_lib --skip_tests     --config Release --build_wheel --update --build --parallel 4    --cmake_generator Ninja     --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) "CMAKE_CUDA_ARCHITECTURES=${CMAKE_CUDA_ARCHITECTURES}" onnxruntime_BUILD_UNIT_TESTS=OFF'

### Error / output

i get error:
[948/1697] Building CXX object CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc.o
FAILED: CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc.o 
/usr/bin/c++ -DCPUINFO_SUPPORTED_PLATFORM=1 -DDISABLE_CUSPARSE_DEPRECATED -DEIGEN_MPL2_ONLY -DEIGEN_USE_THREADS -DENABLE_DLPACK -DENABLE_FP8 -DEXCLUDE_SM_100 -DEXCLUDE_SM_120 -DEXCLUDE_SM_89 -DEXCLUDE_SM_90 -DONLY_C_LOCALE=0 -DONNX_ML=1 -DONNX_NAMESPACE=onnx -DONNX_USE_LITE_PROTO=1 -DORT_ENABLE_STREAM -DPLATFORM_POSIX -DUSE_CUDA=1 -DUSE_FLASH_ATTENTION=1 -DUSE_MEMORY_EFFICIENT_ATTENTION=1 -D_GNU_SOURCE -D__ONNX_NO_DOC_STRINGS -Donnxruntime_providers_cuda_EXPORTS -I/code/include/onnxruntime -I/code/include/onnxruntime/core/session -I/code/build/Linux/Release/_deps/pytorch_cpuinfo-src/include -I/code/build/Linux/Release -I/code/onnxruntime -I/code/build/Linux/Release/_deps/gsl-src/include -I/code/build/Linux/Release/_deps/abseil_cpp-src -I/code/build/Linux/Release/_deps/date-src/include -I/code/build/Linux/Release/_deps/onnx-src -I/code/build/Linux/Release/_deps/onnx-build -I/code/build/Linux/Release/_deps/protobuf-src/src -I/code/build/Linux/Release/_deps/flatbuffers-src/include -I/code/build/Linux/Release/_deps/cutlass-src/include -I/code/build/Linux/Release/_deps/cutlass-src/examples -I/code/build/Linux/Release/_deps/cutlass-src/tools/util/include -I/code/build/Linux/Release/_deps/cudnn_frontend-src/include -I/code/build/Linux/Release/_deps/eigen3-src -isystem /code/build/Linux/Release/_deps/safeint-src -isystem /usr/local/cuda/targets/x86_64-linux/include -isystem /code/build/cudnn/include -isystem /code/build/Linux/Release/_deps/mp11-src/include -ffunction-sections -fdata-sections -DCPUINFO_SUPPORTED -O3 -DNDEBUG -fPIC -Wall -Wextra -Wno-deprecated-copy -Wno-nonnull-compare -Wno-reorder -Wno-error=sign-compare -Werror -MD -MT CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc.o -MF CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc.o.d -o CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc.o -c /code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc
/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc: In member function ‘virtual std::optional<bool> onnxruntime::CUDAExecutionProvider::ShouldConvertDataLayoutForOp(std::string_view, std::string_view, onnxruntime::DataLayout) const’:
/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc:324:90: error: unused parameter ‘node_domain’ [-Werror=unused-parameter]
  324 | std::optional<bool> CUDAExecutionProvider::ShouldConvertDataLayoutForOp(std::string_view node_domain,
      |                                                                         ~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc:325:90: error: unused parameter ‘node_op_type’ [-Werror=unused-parameter]
  325 |                                                                         std::string_view node_op_type,
      |                                                                         ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
/code/onnxruntime/core/providers/cuda/cuda_execution_provider.cc:326:84: error: unused parameter ‘target_data_layout’ [-Werror=unused-parameter]
  326 |                                                                         DataLayout target_data_layout) const {
      |                                                                         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[949/1697] Building CXX object CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_stream_handle.cc.o
[950/1697] Building CXX object CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cuda_provider_factory.cc.o
[951/1697] Building CXX object CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/core/providers/cuda/cudnn_common.cc.o
ninja: build stopped: subcommand failed.
Namespace(build_dir='/code/build/Linux', config=['Release'], update=True, build=True, clean=False, parallel=4, target=None, compile_no_warning_as_error=False, build_shared_lib=True, build_apple_framework=False, enable_lto=False, use_cache=False, use_binskim_compliant_compile_flags=False, cmake_extra_defines=[['ONNXRUNTIME_VERSION=1.23.0', 'CMAKE_CUDA_ARCHITECTURES=75;80;86', 'onnxruntime_BUILD_UNIT_TESTS=OFF']], cmake_path='cmake', cmake_generator='Ninja', use_vcpkg=False, use_vcpkg_ms_internal_asset_cache=False, skip_submodule_sync=True, test=False, skip_tests=True, ctest_path='ctest', enable_onnx_tests=False, path_to_protoc_exe=None, fuzz_testing=False, enable_symbolic_shape_infer_tests=False, skip_onnx_tests=False, skip_winml_tests=False, skip_nodejs_tests=False, test_all_timeout='10800', enable_transformers_tool_test=False, build_micro_benchmarks=False, code_coverage=False, enable_training=False, enable_training_apis=False, enable_training_ops=False, enable_nccl=False, nccl_home=None, enable_memory_profile=False, enable_address_sanitizer=False, gen_doc=None, rv64=False, riscv_toolchain_root='', riscv_qemu_path='', android=False, android_abi='arm64-v8a', android_api=27, android_sdk_path='', android_ndk_path='', android_cpp_shared=False, android_run_emulator=False, build_wasm=False, build_wasm_static_lib=False, emsdk_version='4.0.11', enable_wasm_simd=False, enable_wasm_relaxed_simd=False, enable_wasm_threads=False, disable_wasm_exception_catching=False, enable_wasm_api_exception_catching=False, enable_wasm_exception_throwing_override=True, wasm_run_tests_in_browser=False, enable_wasm_profiling=False, enable_wasm_debug_info=False, wasm_malloc=None, emscripten_settings=None, use_full_protobuf=False, use_mimalloc=False, external_graph_transformer_path=None, use_extensions=False, extensions_overridden_path=None, minimal_build=None, include_ops_by_config=None, enable_reduced_operator_type_support=False, disable_contrib_ops=False, disable_ml_ops=False, disable_rtti=False, disable_types=[], disable_exceptions=False, client_package_build=False, enable_pybind=False, build_wheel=True, wheel_name_suffix=None, skip_keras_test=False, build_csharp=False, build_nuget=False, msbuild_extra_options=None, build_java=False, build_nodejs=False, build_objc=False, use_cuda=True, cuda_version=None, cuda_home='/usr/local/cuda', cudnn_home='/code/build/cudnn', enable_cuda_line_info=False, enable_cuda_nhwc_ops=False, disable_cuda_nhwc_ops=False, enable_cuda_minimal_build=False, nvcc_threads=-1, enable_nvtx_profile=False, enable_cuda_profiling=False, use_dnnl=False, dnnl_gpu_runtime='', dnnl_opencl_root='', dnnl_aarch64_runtime='', dnnl_acl_root='', use_openvino=None, use_tensorrt=False, use_tensorrt_builtin_parser=True, use_tensorrt_oss_parser=False, tensorrt_home=None, tensorrt_rtx_home=None, use_nv_tensorrt_rtx=False, use_dml=False, dml_path='', dml_external_project=False, use_nnapi=False, nnapi_min_api=None, use_coreml=False, use_qnn=None, qnn_home=None, use_snpe=False, snpe_root=None, use_vitisai=False, use_armnn=False, armnn_relu=False, armnn_bn=False, armnn_home=None, armnn_libs=None, use_acl=False, acl_home=None, acl_libs=None, no_kleidiai=False, use_rknpu=False, use_cann=False, cann_home=None, use_migraphx=False, migraphx_home=None, use_rocm=False, rocm_version=None, rocm_home=None, use_webnn=False, use_jsep=False, use_webgpu=False, use_external_dawn=False, wgsl_template='static', use_xnnpack=False, use_vsinpu=False, use_azure=False, enable_lazy_tensor=False, ms_experimental=False, enable_msinternal=False, use_lock_free_queue=False, enable_generic_interface=False, allow_running_as_root=True, enable_external_custom_op_schemas=False)
nvcc_threads=1 to ensure memory per thread >= 4GB for available_memory=23476260864 and fmha_parallel_jobs=4
Traceback (most recent call last):
  File "/code/tools/ci_build/build.py", line 2662, in <module>
    sys.exit(main())
  File "/code/tools/ci_build/build.py", line 2563, in main
    build_targets(args, cmake_path, build_dir, configs, num_parallel_jobs, args.target)
  File "/code/tools/ci_build/build.py", line 1374, in build_targets
    run_subprocess(cmd_args, env=env)
  File "/code/tools/ci_build/build.py", line 148, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
  File "/code/tools/python/util/run.py", line 50, in run
    completed_process = subprocess.run(
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/local/bin/cmake', '--build', '/code/build/Linux/Release', '--config', 'Release', '--', '-j4']' returned non-zero exit status 1.

### Visual Studio Version

_No response_

### GCC / Compiler Version

_No response_

--- a/onnxruntime/core/providers/cuda/cuda_execution_provider.cc
+++ b/onnxruntime/core/providers/cuda/cuda_execution_provider.cc
@@ -321,9 +321,9 @@ DataLayout CUDAExecutionProvider::GetPreferredLayout() const {
   return this->IsNHWCPreferred() ? DataLayout::NHWC : DataLayout::NCHW;
 }
 
-std::optional<bool> CUDAExecutionProvider::ShouldConvertDataLayoutForOp(std::string_view node_domain,
-                                                                        std::string_view node_op_type,
-                                                                        DataLayout target_data_layout) const {
+std::optional<bool> CUDAExecutionProvider::ShouldConvertDataLayoutForOp( [[maybe_unused]] std::string_view node_domain,
+                                                                         [[maybe_unused]] std::string_view node_op_type,
+                                                                         [[maybe_unused]] DataLayout target_data_layout) const {
 #if defined(ENABLE_CUDA_NHWC_OPS)
   if (target_data_layout != DataLayout::NHWC) {
     return std::nullopt;

--- a/onnxruntime/core/providers/cuda/cuda_execution_provider.cc
+++ b/onnxruntime/core/providers/cuda/cuda_execution_provider.cc
@@ -321,9 +321,9 @@ DataLayout CUDAExecutionProvider::GetPreferredLayout() const {
   return this->IsNHWCPreferred() ? DataLayout::NHWC : DataLayout::NCHW;
 }
 
-std::optional<bool> CUDAExecutionProvider::ShouldConvertDataLayoutForOp( [[maybe_unused]] std::string_view node_domain,
-                                                                         [[maybe_unused]] std::string_view node_op_type,
-                                                                         [[maybe_unused]] DataLayout target_data_layout) const {
+std::optional<bool> CUDAExecutionProvider::ShouldConvertDataLayoutForOp([[maybe_unused]] std::string_view node_domain,
+                                                                        [[maybe_unused]] std::string_view node_op_type,
+                                                                        [[maybe_unused]] DataLayout target_data_layout) const {
 #if defined(ENABLE_CUDA_NHWC_OPS)
   if (target_data_layout != DataLayout::NHWC) {
     return std::nullopt;

